connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ErrorReporterTest.java

```diff
@@ -146,6 +146,20 @@ public void testReportDLQTwice() {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testCloseDLQ() {
```

Comment:
If I read this correctly, the guarantee here is that if a DLQ reporter is closed, its internal Kafka producer will also be closed.

Do you think it would also make sense to add a test to ensure that when a `WorkerSinkTask` and `WorkerSourceTask`  are shut down, and/or a `RetryWithToleranceOperator` is closed, the underlying `ErrorReporter` instances are closed as well?

Reply:
I found ErrorHandlingTaskTest that seemed to be doing similar testing, and I managed to test the rest of the hierarchy (WorkerSourceTask/WorkerSinkTask -> RetryWithToleranceOperator -> ProcessingContext -> ErrorReporter) in just two tests.

Combined with the DLQ reporter -> KafkaProducer test from the first commit, I think the testing is adequate now.

Reply:
Agreed, looks great!